test/python/circuit/test_circuit_load_from_qasm.py

```diff
@@ -108,9 +130,28 @@ def test_loading_all_qelib1_gates(self):
         ref_circuit.crz(0.6, 0, 1)
         ref_circuit.rxx(0.2, 0, 1)
         ref_circuit.rzz(0.2, 0, 1)
-        ref_circuit.measure([0, 1, 2], [0, 1, 2])
 
-        self.assertEqual(qasm_circuit, ref_circuit)
+        ref_circuit.rccx(0, 1, 2)
+        ref_circuit.rcccx(0, 1, 2, 3)
+        ref_circuit.append(C3XGate(), [0, 1, 2, 3])
+        ref_circuit.append(C3SXGate(), [0, 1, 2, 3])
+        ref_circuit.append(C4XGate(), [0, 1, 2, 3, 4])
+
+        ref_circuit.measure_all()
+
+        # check that ref_circuit contains all the operations from qasm_circuit
+        ref_ops = ref_circuit.count_ops().keys()
+        qasm_ops = qasm_circuit.count_ops().keys()
+        self.assertTrue(qasm_ops <= ref_ops, msg=qasm_ops - ref_ops)
+
+        # check that all qelib1.inc gates are in ref_circuit
+        qelib1_gates = self.get_qelib1_gates()
+        qelib1_gates -= {"u0"}  # u0 is not supported outside of qelib1
+        qelib1_gates -= {"rc3x"}  # rc3x is defined as rcccx in RC3XGate class
+        qelib1_gates -= {"c3x"}  # c3x is defined as mcx in C3XGate class
+        qelib1_gates -= {"c4x"}  # c4x is defined as mcx in C4XGate class
+
+        self.assertTrue(qelib1_gates <= ref_ops, msg=qelib1_gates - ref_ops)
```

Comment:
I'm not 100% sure what the purpose of this test is.  At any rate, `qelib1.inc` should be fixed, and not subject to additions and removals (we weren't great at that in the past), so I think it may not be necessary.